storage/src/main/java/org/apache/kafka/storage/internals/log/LogSegment.java

```diff
@@ -805,6 +805,10 @@ private Void deleteTypeIfExists(StorageAction<Boolean, IOException> delete, Stri
                 if (logIfMissing) {
                     LOGGER.info("Failed to delete {} {} because it does not exist.", fileType, file.getAbsolutePath());
                 }
+                if (file.getParent() == null) {
+                    LOGGER.warn("Failed to delete {} {} because parent directory is null.", fileType, file.getAbsolutePath());
```

Comment:
I don't think we have to log anything here. We can assume the file is already deleted.

Reply:
Hi @showuon, thanks for the suggestion. I removed the log and add a test with mock for it.